types/time.go

```diff
@@ -559,6 +559,46 @@ func TimestampDiff(unit string, t1 Time, t2 Time) int64 {
 func ParseDateFormat(format string) []string {
 	format = strings.TrimSpace(format)
 
+	start := 0
+	var seps = make([]string, 0)
+	for i := 0; i < len(format); i++ {
+		if len(seps) > 7 { //max len is 7,'Y-M-D H-M-S.ms'
+			break
+		}
+		// Date format must start with number.
+		if i == 0 {
```

Comment:
If put it outside the loop,I will need to judge the format is empty or not. How do u think?

Reply:
You should move it out because
1. The code will be clean
2. The performance will be better

```
for i := 0; i <len(xx); i++ {
      if i == 0 { 
          // This check runs many times,
          // but when i != 0, the check is pointless.
      }
}
```

```
if len(xx) == 0 {
    return
}
handle(xx[0])
for i := 1; i < len(xx); i++ {
    handle(xx[i]) 
}  
```